protocol.md

```diff
@@ -0,0 +1,249 @@
+
+Whisper's Proof Of Work algorithm is used to to deter denial of service and various spam/flood attacks against the Whisper network. The sender of a message must perform some work which in this case means performing some computations. Because Status main client is a mobile client, this easily leads to battery draining and poor performance of the app itself. Hence, these clients send envelopes with the following settings:
+* proof-of-work equal `0.002`
+* proof-of-work time equal to `10`
+* time-to-live equal `10`
+
+If you want to receive messages from a mobile client, you need to run Whisper node with PoW set to `0.002`. In case of `geth`, this option can be overriden with `-shh.pow=0.002` flag.
+
+TODO: provide an instruction how to start a Whisper node with proper configuration using geth.
+
+## Keys management
+
+TBD
+
+## Encryption algorithms
+
+TBD (link to Whisper spec)
+
+## Topic
+
+There are two types of Whisper topics the protocol uses:
+* static topic for private chats (also called _contact discovery topic_)
+* dynamic topic based on a chat name for public chats
+
+The static topic is always the same and its hex representation is `0xf8946aac`. In fact, _the contact discovery topic_ is calculated using a dynamic topic algorithm described below with a constant name `contact-discovery`.
+
+Having only one topic for all private chats has an advantage as it's very hard to figure out who talks to who. A drawback is that everyone receives everyone's messages but they can decrypt only these they have private keys for.
+
+A dynamic topic is derived from a string using the following algorithm:
+
+```
+var hash []byte
+
+hash = keccak256(name)
+
+# Whisper V6 specific
+var topic [4]byte
+
+topic_len = 4
+
+if len(hash) < topic_len {
+    topic_len = len(hash)
+}
+
+for i = 0; i < topic_len; i++ {
+    topic[i] = hash[i]
+}
+```
+
+## Encryption
+
+Protocol distinguishes messages encrypted using asymmetric and symmetric encryptions.
+
+Symmetric keys are created using `shh_generateSymKeyFromPassword` Whisper JSON-RPC method which accepts one param, a string.
+
+Messages encrypted with asymmetric encryption should be encrypted using recipeint's public key so that only the recipient can decrypt them.
+
+# Perfect Forward Secrecy (PFS)
+
+TODO: link to a separate document (currently in the PR).
+
+[PFS in Status.im docs](https://status.im/research/pfs.html)
+
+# Device syncing
+
+TODO: link to a separate document.
+
+# One-to-one messages
+
+One-to-one messages are also known as private messages.
+
+TODO: describe how to send a 1-1 message starting from adding a key in Whisper etc.
+
+# Public messages
+
+TODO: describe how to send a public message starting from adding a key in Whisper etc.TBD
+
+# Group messages
+
+TODO: describe how to send a group message starting from adding a key in Whisper etc.
+
+# Offline messages
```

Comment:
Maybe this section can be changed to be just a "whisper node heirarchy" that details the specifics of the different types of clients that exist on the network, their particular use-case, load, and required trust.
- mobile
- whisper node
- mailserver
- ULC and LES ??

Reply:
I am not sure we want to do that. I like the idea that each Whisper node is the same, however, in fact we have light Whisper clients and mail servers so it makes sense. I would keep this section but create another one which explains these slight differences between various nodes. Does that sound good?